main.cpp

```diff
@@ -147,89 +130,68 @@ struct FloatType
     FloatType(float f) : value(new float(f)) {}
     ~FloatType()
     {
+        value = nullptr;
         delete value;
```

Comment:
you have a memory leak now
you are deleting `nullptr`, not `value`'s memory address.
- [ ] revise